### ISSUE TITLE
Segment Anything - Next Steps

### ISSUE THREAD

**Issue by tirthasheshpatel** [keras-team/keras-cv#2081]
### Segment Anything model in KerasCV

#1987 added the Segment Anything model to KerasCV but it needs a guide to demonstrate training and prompting the trained model. It would also be nice to have some benchmarks for it. Following is a list of tasks that still need to be addressed:

- [x] Writing a guide to use the model (similar to the [predictor demo](https://github.com/facebookresearch/segment-anything/blob/6fdee8f2727f4506cfbbe553e23b895e27956588/notebooks/predictor_example.ipynb) on the original repo)
- [ ] Writing a guide to generate masks for the entire image (similar to the [automatic mask generation demo](https://github.com/facebookresearch/segment-anything/blob/6fdee8f2727f4506cfbbe553e23b895e27956588/notebooks/automatic_mask_generator_example.ipynb) on the original repo)
- [ ] (@IMvision12) A guide for end-to-end model training. See the appendix "A. Segment Anything Model and Task Details" from [the paper](https://arxiv.org/pdf/2304.02643.pdf) that explains the training method used for the existing presets.
- [ ] Adding support for text prompts (e.g. CLIP). The paper mentions that a CLIP model was used to encode text prompts.
- [x] (Optionally) It'd also be nice to have benchmarks for all the backends and document it somewhere.

**Comment by IMvision12** [comment]
@tirthasheshpatel i can take up end - to - end model training guide

**Comment by AnimeshMaheshwari22** [comment]
@tirthasheshpatel I can take the Writing a guide to use the model

**Comment by cosmo3769** [comment]
@tirthasheshpatel I can take up "Writing a guide to generate masks for the entire image". Thank you.

**Comment by cosmo3769** [comment]
> @tirthasheshpatel I can take up "Writing a guide to generate masks for the entire image". Thank you.

@tirthasheshpatel Since this guide requires implementation of `predictor` and `maskgenerator`, I am working on this.

**Comment by cosmo3769** [comment]
@tirthasheshpatel Since the `predictor` and `automaticmaskgenerator`  work is in wip and almost done by you as I checked your repo here: https://github.com/tirthasheshpatel/segment_anything_keras/tree/keras-cv-update. If there is any help required, I would be happy to help. Thank you.

**Comment by sachinprasadhs** [comment]
Segment anything is now available in Keras Hub https://keras.io/keras_hub/guides/segment_anything_in_keras_hub/
This issue can be closed
